app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,236 @@
+        }
+        $activationCode = rand(100000, 999999);
+        $userData = [
+            'first_name' => $requestData['first_name'],
+            'last_name' => $requestData['last_name'] ?? null,
+            'email' => $requestData['email'],
+            'password' => bcrypt($requestData['password']),
+            'phone' => $requestData['phone'] ?? null,
+            'gender' => $requestData['gender'] ?? null,
+            'address' => $requestData['address'] ?? null,
+            'city' => $requestData['city'] ?? null,
+            'country' => $requestData['country'] ?? null,
+            'user_type' => $requestData['user_type'],
+            'activation_code' => $activationCode,
+        ];
+        $userInfo = AuthRepository::save($userData);
+        if (!$userInfo instanceof User) {
+            return response()->json(['status' => 500, 'message' => 'Cannot register user'], 200);
+        }
+        if ($userInfo['user_type'] == UserType::Company) {
+            $companyData = [
+                'company_name' => $requestData['company_name'],
+                'company_size' => $requestData['company_size'],
+                'company_address' => $requestData['company_address'],
+                'company_city' => $requestData['company_city'],
+                'company_country' => $requestData['company_country'],
+                'user_id' => $userInfo['id'],
+            ];
+            $companyInfo = CompaniesRepository::save($companyData);
+            if (!$companyInfo instanceof Companies) {
+                User::where('id', $userInfo['id'])->forceDelete();
+                return response()->json(['status' => 500, 'message' => 'Cannot save company'], 200);
+            }
+            $userInfo->company_id = $companyInfo['id'];
+            if (!$userInfo->save()) {
+                User::where('id', $userInfo['id'])->forceDelete();
+                Companies::where('id', $companyInfo['id'])->forceDelete();
+                return response()->json(['status' => 500, 'message' => 'Cannot register user'], 200);
+            }
+        }
+        AuthRepository::sendVerificationEmail($userInfo);
+        return response()->json(['status' => 200, 'message' => 'A verification mail has been sent to your email, Please verify the email to login'], 200);
+    }
+
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function verifyAccount(Request $request): JsonResponse
+    {
+        $requestData = $request->all();
+        $validator = Validator::make($requestData, [
+            'code' => 'required|string',
+            'email' => 'required|email',
+        ]);
+        if ($validator->fails()) {
+            return response()->json(['status' => 500, 'errors' => $validator->errors()]);
+        }
+        $userInfo = User::where('email', $requestData['email'])->where('activation_code', base64_decode($requestData['code']))->first();
```

Comment:
When the code is getting base64_encode so that we have to decode here ? Bit of confused.